File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/presentation/common/Presenter.java
package com.crpr.androidcinema.presentation.common;

import rx.subscriptions.CompositeSubscription;

/**
 * Created by claudioribeiro on 08/07/16.
 */
public abstract class Presenter implements Base.Presenter {

    protected boolean _isMakingRequest;
    protected CompositeSubscription _subscriptions = new CompositeSubscription();

    //This methods should map to activities lifecycle

    public void onCreate(){}

    public void onStart(){}

    public void onStop(){}

    public void onPause(){}

    public void onDestroy(){
        if(_subscriptions != null && _subscriptions.hasSubscriptions()){
            _subscriptions.unsubscribe();
            _subscriptions = null;
        }
    }

    public abstract void bindView(Base.View view);

}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/presentation/common/Presenter.kt
package com.crpr.androidcinema.presentation.common

import rx.subscriptions.CompositeSubscription

/**
 * Created by claudioribeiro on 08/07/16.
 */
abstract class Presenter : Base.Presenter {

    protected var _isMakingRequest: Boolean = false
    protected var _subscriptions: CompositeSubscription? = CompositeSubscription()

    //This methods should map to activities lifecycle

    open fun onCreate() {}

    override fun onStart() {}

    fun onStop() {}

    override fun onPause() {}

    override fun onDestroy() {
        if (_subscriptions != null && _subscriptions!!.hasSubscriptions()) {
            _subscriptions!!.unsubscribe()
            _subscriptions = null
        }
    }

    abstract override fun bindView(view: Base.View)

}
